tcms/static/js/testrun_actions.js

```diff
@@ -259,91 +269,61 @@ function updateRunStatus(object_pk, value, callback) {
   });
 }
 
-var updateCaseRunStatus = function(e) {
-  e.stopPropagation();
-  e.preventDefault();
-  var container = jQ(this).parents().eq(3);
-  var parent = container.parent();
-  var title = parent.prev();
-  var link = title.find('.expandable')[0];
-  var parameters = Nitrate.Utils.formSerialize(this);
-  var object_pk = parameters['object_pk'];
-  var value = parameters['value'];
-
-  // Callback when
-  var callback = function(t) {
-    // Update the contents
-    const value = parameters['value']
-    if (value != '') {
-      // Update the case run status icon
-      var crs = Nitrate.TestRuns.CaseRunStatus;
-      title.find('.execution_status_icon').each(function(index) {
-        for (i in crs) {
-          jQ(this).removeClass(crs[i].icon);
-        }
-        const execution = Nitrate.TestRuns.CaseRunStatus.find(tcs => tcs.pk === value);
-        jQ(this).addClass(execution.icon);
-        jQ(this).css('color', execution.color)
-      });
+function updateExecutionStatus(executionId, caseId, executionStatusPk) {
 
-      // Update related people
-      var usr = Nitrate.User;
-      title.find('.link_tested_by').each(function(i) {
-        jQ(this).html(usr.username);
-      });
-    }
+  const container = jQ('.update_form').parents().eq(3)
+  const parent = container.parent()
+  const title = parent.prev()
+  const link = title.find('.expandable')[0];
+
+  const comment = jQ('td > textarea[name=comment]').val()
+
+  setContainerToLoading(container, caseId)
```

Comment:
I don't see a reason to do this. Setting a spinner only to remove it in the next few lines. IMO not needed.